### PR TITLE
Simplify IP address logic.

### PR DESCRIPTION
**Pull request by mrstegeman**
Match the reported websocket href to the requested host.

### CODE REVIEW

**Review by hobinjk — APPROVED** [review]

**Comment by mrstegeman** [comment]
@freaktechnik ping

File: src/main/java/org/mozilla/iot/webthing/Utils.java [hunk 1]
```diff
@@ -30,4 +24,32 @@ public static String timestamp() {
...
+     * Get all IP addresses.
+     *
+     * @return List of addresses.
+     */
+    public static List<String> getAddresses() {
+        Set<String> addresses = new HashSet<>();
+        try {
+            for (NetworkInterface iface : Collections.list(NetworkInterface.getNetworkInterfaces())) {
+                for (InetAddress address : Collections.list(iface.getInetAddresses())) {
+                    String s = address.getHostAddress().split("%")[0];
```
Comment (hobinjk) [inline src/main/java/org/mozilla/iot/webthing/Utils.java]: A comment explaining how these lines work would be nice for the next person to look at it

File: src/main/java/org/mozilla/iot/webthing/WebThingServer.java [hunk 2]
```diff
@@ -98,18 +98,20 @@ public WebThingServer(ThingsType things,
...
-        this.ip = Utils.getIP();
         this.isTls = sslOptions != null;
         this.hostname = hostname;
 
         this.hosts = new ArrayList<>();
-        this.hosts.add("127.0.0.1");
-        this.hosts.add(String.format("127.0.0.1:%d", this.port));
         this.hosts.add("localhost");
         this.hosts.add(String.format("localhost:%d", this.port));
         this.hosts.add(this.ip);
```
Comment (hobinjk) [inline src/main/java/org/mozilla/iot/webthing/WebThingServer.java]: It looks like this.ip will be null here

Comment (mrstegeman) [reply src/main/java/org/mozilla/iot/webthing/WebThingServer.java]: Fixed.

File: src/main/java/org/mozilla/iot/webthing/Utils.java (hunk 1 above)
Comment (mrstegeman) [reply src/main/java/org/mozilla/iot/webthing/Utils.java]: Done.

**Review by hobinjk — APPROVED** [review]

**Pull request merged**